onnxscript/tests/models/signal_dft.py

```diff
@@ -162,16 +153,14 @@ def dft_last_axis(
     one = op.Constant(value=make_tensor("one", TensorProto.INT64, [1], [1]))
     two = op.Constant(value=make_tensor("two", TensorProto.INT64, [1], [2]))
     last = op.Constant(value=make_tensor("last", TensorProto.INT64, [1], [-1]))
-    shape1 = op.Constant(value=make_tensor("shape1", TensorProto.INT64, [2], [-1, 1]))
-    shape2 = op.Constant(value=make_tensor("shape2", TensorProto.INT64, [2], [1, -1]))
 
-    nar = op.Range(zero, fft_length, one)  # fft_length or dim
-    n0 = op.Cast(nar, to=1)
-    n = op.Reshape(n0, shape1)
+    range = op.Range(zero, fft_length, one)  # fft_length or dim
+    range_float = op.Cast(range, to=1)
+    shape1 = op.Constant(value=make_tensor("shape1", TensorProto.INT64, [2], [-1, 1]))
+    n = op.Reshape(range_float, shape1)
 
-    kar = op.Range(zero, fft_length, one)
-    k0 = op.Cast(kar, to=1)
-    k = op.Reshape(k0, shape2)
+    shape2 = op.Constant(value=make_tensor("shape2", TensorProto.INT64, [2], [1, -1]))
```

Comment:
It would be nice if we can make tensor w/o needing to supply a name

Reply:
Agree. Ideally, we should be able to use just `[1, -1]` and have the converter automatically do the rest. It almost works now, but there are a couple of issues. Eager-mode doesn't seem to fully handle it. But, there's also a design-issue: how to map a value such as `[1, -1]` to an ONNX value. A 1-dimensional tensor is the most-common and useful choice, but there could be rare cases where we want a Sequence of scalar tensors. So, a complete solution requires a bit more thought.

Reply:
Personally I like an np.array() like interface. So we don't confuse a tensor with a python sequence